syntax = "proto3";
package pb.socket;

// 
message HelloReq {
  string name = 1;

}

// 
message UserInfo {
  int64 id = 1;
  string name = 2;
  string jwt = 3;

}

// 对外服数据协议
message ExternalMessage {
  // 请求命令类型: 0 心跳，1 业务
  int32 cmdCode = 1;
  // 协议开关，用于一些协议级别的开关控制，比如 安全加密校验等。 : 0 不校验
  int32 protocolSwitch = 2;
  // 业务路由（高16为主, 低16为子）
  int32 cmdMerge = 3;
  // 响应码。
  int32 responseStatus = 4;
  // 验证信息
  string validMsg = 5;
  bytes data = 6;

}

// 
message HelloSpringMsg {
  string name = 1;

}

// 
message DemoLoginVerify {
  // jwt
  string jwt = 1;

}

// 
message UserLogin {
  string username = 1;
  string password = 2;

}

